addons/web_tour/static/src/js/tip.js

```diff
@@ -156,41 +235,76 @@ var Tip = Widget.extend({
+        let $parent = this.$ideal_location;
+        if ($parent.is('html,body') && this.viewPortState !== "in") {
+            this.$el.css({position: 'fixed'});
+        }
+        const paddingTop = parseInt($parent.css('padding-top'));
+        const paddingLeft = parseInt($parent.css('padding-left'));
+        const paddingRight = parseInt($parent.css('padding-right'));
+
+        const center = (paddingLeft + paddingRight) + ((($parent[0].clientWidth - (paddingLeft + paddingRight)) / 2) - this.$el[0].offsetWidth / 2);
+        const topPosition = $parent[0].offsetTop;
+        let top;
+        if (this.viewPortState === 'up') {
+            top = topPosition + this.$el.innerHeight() + paddingTop;
+            this.$el.css({top: top, left: center});
+        } else if (this.viewPortState === 'down') {
+            top = topPosition + $parent.innerHeight() - this.$el.innerHeight() * 2;
+            this.$el.css({top: top, left: center});
+        } else {
+            this.$el.position({
+                my: this._get_spaced_inverted_position(appendAt),
+                at: appendAt,
+                of: this.$anchor,
+                collision: "none",
+            });
+        }
 
         // Reverse overlay if direction is right to left
         var positionRight = _t.database.parameters.direction === 'rtl' ? "right" : "left";
         var positionLeft = _t.database.parameters.direction === 'rtl' ? "left" : "right";
-        var offset = this.$el.offset();
+
+        // get the offset position of this.$el
+        // Couldn't use offset() or position() because their values are not the desired ones in all cases
+        const offset = {top: this.$el[0].offsetTop, left: this.$el[0].offsetLeft};
         this.$tooltip_overlay.css({
             top: -Math.min((this.info.position === "bottom" ? this.info.space : this.info.overlay.y), offset.top),
             right: -Math.min((this.info.position === positionRight ? this.info.space : this.info.overlay.x), this.$window.width() - (offset.left + this.init_width + this.double_border_width)),
             bottom: -Math.min((this.info.position === "top" ? this.info.space : this.info.overlay.y), this.$window.height() - (offset.top + this.init_height + this.double_border_width)),
             left: -Math.min((this.info.position === positionLeft ? this.info.space : this.info.overlay.x), offset.left),
         });
-
-        this.position = this.$el.position();
+        this.position = offset;
 
         this.$el.addClass("o_animated");
     },
     _bind_anchor_events: function () {
         this.consume_event = Tip.getConsumeEventType(this.$anchor);
+        if (this.info.consumeAction === "input") {
+            var contentEditable = this.$anchor.closest("[contenteditable='true']");
+            if (this.consume_event !== this.info.consumeAction && contentEditable.length) {
+                this.consume_event = "contenteditable";
+            }
+        }
```

Comment:
... what is this seriously?
So now `this.info.consumeAction` can just be "input" otherwise you don't even use it?
Do what I said earlier: in case of "input" consume_event, take the contenteditable element only if not an input or textarea...

Reply:
I didn't understand it like that before, now is more clear, I'll do it